src/controls/controls.actions.ts

```diff
@@ -268,138 +270,116 @@ import { renderCircleControl, renderSquareControl } from './controls.render';
-      );
-      // now we have to find the sign of the skew.
-      // it mostly depend on the origin of transformation.
-      if (transform.originX === LEFT && transform.originY === BOTTOM) {
-        newSkew = -newSkew;
-      }
-      if (transform.originX === RIGHT && transform.originY === TOP) {
-        newSkew = -newSkew;
-      }
-      if (targetHasOneFlip(target)) {
-        newSkew = -newSkew;
-      }
-    }
-    var hasSkewed = currentSkew !== newSkew;
-    if (hasSkewed) {
-      var dimBeforeSkewing = target._getTransformedDimensions().y;
-      target.set('skewX', newSkew);
-      compensateScaleForSkew(target, 'skewY', 'scaleY', 'y', dimBeforeSkewing);
+  function skewHandler(axis: TAxis, eventData, transform, x, y) {
+    const { target } = transform,
+      {
+        counterAxis,
+        origin: originKey,
+        lockSkewing: lockSkewingKey,
+        skew: skewKey,
+      } = AXIS_KEYS[axis];
+    if (target[lockSkewingKey]) {
+      return false;
     }
-    return hasSkewed;
-  }
 
-  /**
-   * Action handler for skewing on the Y axis
-   * @private
-   */
-  function skewObjectY(eventData, transform, x, y) {
-    var target = transform.target,
-      // find how big the object would be, if there was no skewX. takes in account scaling
-      dimNoSkew = target._getTransformedDimensions({
-        skewX: target.skewX,
-        skewY: 0,
-      }),
-      localPoint = getLocalPoint(
-        transform,
-        transform.originX,
-        transform.originY,
-        x,
-        y
-      ),
-      // the mouse is in the center of the object, and we want it to stay there.
-      // so the object will grow twice as much as the mouse.
-      // this makes the skew growth to localPoint * 2 - dimNoSkew.
-      totalSkewSize = Math.abs(localPoint.y * 2) - dimNoSkew.y,
-      currentSkew = target.skewY,
-      newSkew;
-    if (totalSkewSize < 2) {
-      // let's make it easy to go back to position 0.
-      newSkew = 0;
-    } else {
-      newSkew = radiansToDegrees(
-        Math.atan2(totalSkewSize / target.scaleY, dimNoSkew.x / target.scaleX)
-      );
-      // now we have to find the sign of the skew.
-      // it mostly depend on the origin of transformation.
-      if (transform.originX === LEFT && transform.originY === BOTTOM) {
-        newSkew = -newSkew;
-      }
-      if (transform.originX === RIGHT && transform.originY === TOP) {
-        newSkew = -newSkew;
-      }
-      if (targetHasOneFlip(target)) {
-        newSkew = -newSkew;
-      }
-    }
-    var hasSkewed = currentSkew !== newSkew;
-    if (hasSkewed) {
-      var dimBeforeSkewing = target._getTransformedDimensions().x;
-      target.set('skewY', newSkew);
-      compensateScaleForSkew(target, 'skewX', 'scaleX', 'x', dimBeforeSkewing);
-    }
-    return hasSkewed;
+    const counterOriginFactor = new Point(
+        target.resolveOriginX(transform.originX),
+        target.resolveOriginY(transform.originY)
```

Comment:
once #8390 is in will need to refactor this bit

Reply:
also this seems silly to resolve both origin when we know we are going to get one only. it depends on counterAxis we should be able to figure it out.

Reply:
like just adding another key to that object and avoiding creating the point